pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go

```diff
@@ -841,3 +842,52 @@ func TestValidateArrayResultsIndex(t *testing.T) {
 		})
 	}
 }
+
+func Test_paramValueFromCustomRunResult(t *testing.T) {
+	type args struct {
+		result string
+	}
+	tests := []struct {
+		name string
+		args args
+		want *v1.ParamValue
+	}{
+		{
+			name: "multiple array elements result",
+			args: args{
+				result: `["amd64", "arm64"]`,
+			},
+			want: &v1.ParamValue{
+				Type:     "array",
+				ArrayVal: []string{"amd64", "arm64"},
+			},
+		},
+		{
+			name: "single array elements result",
+			args: args{
+				result: `[ "amd64" ]`,
+			},
+			want: &v1.ParamValue{
+				Type:     "array",
+				ArrayVal: []string{"amd64"},
+			},
+		},
+		{
+			name: "simple string result",
+			args: args{
+				result: "amd64",
+			},
+			want: &v1.ParamValue{
+				Type:      "string",
+				StringVal: "amd64",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := paramValueFromCustomRunResult(tt.args.result); !reflect.DeepEqual(got, tt.want) {
```

Comment:
We use `cmp` for comparisons (see [here](https://github.com/tektoncd/pipeline/blob/68336fc53578516f33dd3aff5f22dff3fb2e9ed4/pkg/reconciler/pipelinerun/resources/resultrefresolution_test.go#L540) for example)

Reply:
Thanks for the feedback! I've updated the comparison to use cmp as per the project's coding conventions. Please take a look at the latest commit.